File: python-cli/src/convex_notes/ui/table.py
"""Table display for notes."""

from rich.console import Console
from rich.table import Table

from convex_notes.entity import Note

console = Console()


def display_notes(notes: list[Note]) -> None:
    """Display notes in a formatted table."""
    table = Table(
        show_header=True,
        header_style="bold",
        border_style="dim",
    )

    table.add_column("ID", style="cyan", no_wrap=True)
    table.add_column("Title", style="yellow")
    table.add_column("Content", style="white")
    table.add_column("Created", style="magenta")

    for note in notes:
        table.add_row(
            note.short_id,
            note.title,
            note.content_preview,
            note.formatted_created,
        )

    console.print(table)

File: backend/crates/cli/src/ui/table.rs
//! Table display utilities.

use prettytable::{Cell, Row, Table, format};

use common::format_timestamp;
use db::Note;

/// Display notes in a formatted table.
pub fn display_notes(notes: &[Note]) {
    let mut table = Table::new();
    table.set_format(*format::consts::FORMAT_BOX_CHARS);

    table.add_row(Row::new(vec![
        Cell::new("ID").style_spec("bFc"),
        Cell::new("Title").style_spec("bFy"),
        Cell::new("Content").style_spec("bFw"),
        Cell::new("Created").style_spec("bFm"),
    ]));

    for note in notes {
        table.add_row(Row::new(vec![
            Cell::new(note.short_id()).style_spec("Fc"),
            Cell::new(&note.title).style_spec("Fy"),
            Cell::new(&note.content_preview()),
            Cell::new(&format_timestamp(note.created_at)).style_spec("Fm"),
        ]));
    }

    table.printstd();
}
